### PR TITLE
Correct the wrong serial number

### PR DESCRIPTION
**Pull request by hi-rustin**
the number of the parts should be seven if we count "Appendices" as one.

### CODE REVIEW

File: src/about-this-guide.md [hunk 1]
```diff
@@ -9,7 +9,8 @@ There are six parts to this guide:
 1. [Building and debugging `rustc`][p1]: Contains information that should be
    useful no matter how you are contributing, about building, debugging,
    profiling, etc.
-2. [Contributing to `rustc`][p1-5]: Contains information that should be useful
+   
+   [Contributing to `rustc`][p1-5]: Contains information that should be useful
```
Comment (JohnTitor) [inline src/about-this-guide.md:13]: Hmm, the number of the parts should be seven if we count "Appendices" as one.

**Review by JohnTitor — CHANGES_REQUESTED** [review]

**Comment by JohnTitor** [comment]
Also, could you update the [p1-p5] link to ./contributing.html?

File: src/about-this-guide.md (hunk 1 above)
Comment (hi-rustin) [reply src/about-this-guide.md]: Fixed.

File: src/about-this-guide.md [hunk 2]
```diff
@@ -4,26 +4,26 @@ This guide is meant to help document how rustc – the Rust compiler –
...
 development.
 
-There are six parts to this guide:
+There are seven parts to this guide:
 
 1. [Building and debugging `rustc`][p1]: Contains information that should be
    useful no matter how you are contributing, about building, debugging,
    profiling, etc.
-2. [Contributing to `rustc`][p1-5]: Contains information that should be useful
+2. [Contributing to `rustc`][contributing]: Contains information that should be useful
```
Comment (JohnTitor) [inline src/about-this-guide.md:12]: nit: I'd like to keep the consistency of the links here. I mean, the chapter name vs. the chapter number. I'm fine either way.

Comment (hi-rustin) [reply src/about-this-guide.md:12]: Actually I'm not quite sure which chapter p1-5 is referring to? I'm not quite sure what  chapter number to use in this place.

Comment (JohnTitor) [reply src/about-this-guide.md:12]: I guess it was considered as a subchapter? It explains why it said "six chapters". But now it has its own chapter so it should make sense to call it p2.

Comment (hi-rustin) [reply src/about-this-guide.md]: Fixed.

**Review by JohnTitor — APPROVED** [review]
Thanks! Still, part-n-intro.md is confusing but it's out of scope on this PR and we can discuss on another issue.

**Pull request merged**